src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,238 @@
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer_id());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in driversDB.", e);
+        }
+        deleteAllDriversFromCar(car.getId());
+        insertDriversToCar(car);
+        return car;
+    }
+
+    private void deleteAllDriversFromCar(Long id) {
+        String query = "DELETE FROM cars_drivers "
+                + "WHERE car_id = ? AND driver_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteDriversStatement
+                        = connection.prepareStatement(query)) {
+            deleteDriversStatement.setLong(1, id);
+            List<Long> driversId = new ArrayList<>();
+            driversId = getAllDriversIdByCar(id);
+            for (Long driverId : driversId) {
+                deleteDriversStatement.setLong(2, driverId);
+                deleteDriversStatement.execute();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t delete drivers from car №" + id, e);
+        }
+    }
+
+    public List<Long> getAllDriversIdByCar(Long carId) {
+        String getAllOriversForCar = "SELECT driver_id FROM cars_drivers "
+                + "WHERE car_id = ?;";
+        List<Long> ids = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement =
+                        connection.prepareStatement(getAllOriversForCar)) {
+            getAllCarsStatement.setLong(1, carId);
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                ids.add(resultSet.getObject("driver_id", Long.class));
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't find drivers in DB by car id=" + carId, e);
+        }
+        return ids;
+    }
+
+    private void insertDriversToCar(Car car) {
+        String insertDriversRequest = "INSERT INTO `cars_drivers` "
+                + "(car_id, driver_id) VALUE (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertDriversStatement =
+                        connection.prepareStatement(insertDriversRequest)) {
+            insertDriversStatement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                insertDriversStatement.setLong(2, driver.getId());
+                insertDriversStatement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t insert drivers to " + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
```

Comment:
public methods should be placed above private

Reply:
removed